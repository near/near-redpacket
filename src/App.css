
* {
  font-family: 'Inter', sans-serif;
}
body {
  margin: 0;
}

.App-header {
  background-image: url('https://nearprotocol.com/wp-content/uploads/2019/03/illo-developers.svg');
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  color: white;
  text-align: center;
}

.App-link {
  color: #61dafb;
}

.logo-wrapper {
  display: flex;
  flex-direction: row-reverse;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  pointer-events: none;
}

.margin-logo {
  margin: 20px;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.image-wrapper {
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: center;
}

.logo {
  width: 300px;
}

p {
  color: #25282A;
}

.login {
  margin-top: 32px;
}

button {
  padding: 8px;
  color: #0072CE;
  border-radius: 32px;
  border: 2px solid #0072CE;
  margin: 8px 8px 16px 8px;
  font-size: 1rem;
  outline: none;
}

button:hover {
  cursor: pointer;
  color: white;
  background: #0072CE;
  outline: none;
} 

