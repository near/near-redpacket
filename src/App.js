import 'regenerator-runtime/runtime';
import React, { Component } from 'react';
import Drops from './Drops';
import nearlogo from './assets/near-logo.svg';
import iconUser from './assets/img/icon-account.svg';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      login: false,
      currentUser: window.currentUser,
    }
    this.signedInFlow = this.signedInFlow.bind(this);
    this.requestSignIn = this.requestSignIn.bind(this);
    this.requestSignOut = this.requestSignOut.bind(this);
    this.signedOutFlow = this.signedOutFlow.bind(this);
    this.updateUser = this.updateUser.bind(this);
  }

  async componentDidMount() {
    let loggedIn = this.props.wallet.isSignedIn();
    if (loggedIn) {
      this.signedInFlow();
    } else {
      this.signedOutFlow();
    }
    this.setState({ currentUser: window.currentUser })
  }

  async signedInFlow() {
    this.setState({
      login: true,
    })
    const accountId = await this.props.wallet.getAccountId()
    if (window.location.search.includes("account_id")) {
      window.location.replace(window.location.origin + window.location.pathname)
    }
  }

  async requestSignIn() {
    const appTitle = 'NEAR Red Envelope';
    await this.props.wallet.requestSignIn(
      window.nearConfig.contractName,
      appTitle
    )
  }

  async updateUser() {
    await window.getCurrentUser()
    this.setState({ currentUser: window.currentUser })
  }

  requestSignOut() {
    this.props.wallet.signOut();
    setTimeout(this.signedOutFlow, 500);
    console.log("after sign out", this.props.wallet.isSignedIn())
  }

  signedOutFlow() {
    if (window.location.search.includes("account_id")) {
      window.location.replace(window.location.origin + window.location.pathname)
    }
    this.setState({
      login: false,
      currentUser: null,
    })
  }

  render() {
    const {
      state,
      updateUser
    } = this
    const {
      currentUser
    } = state

    console.log(state)

    return (
      <div className="near-container">
        <div className="near-dapp">
          <div className="near-dapp-header">
            <div className="near-logo">
              <img className="near-logo" src={nearlogo} alt="NEAR logo" height="32" />
            </div>
            <div className="near-user">
              {this.state.login ? 
                  <div className="dropdown" tabIndex="0">
                      <div className="btn">
                          <img className="btn-icon" src={iconUser} alt="NEAR user" height="40" />
                          <span className="text-ellipsis">{currentUser.account_id}</span>
                      </div>
                      <ul className="menu">
                          <li className="menu-item">
                              <a href="#" onClick={this.requestSignOut}>
                                  Log Out
                              </a>
                          </li>
                      </ul>
                  </div>
                  : <a className="btn" href="#" onClick={this.requestSignIn}>
                      <img className="btn-icon" src={iconUser} alt="NEAR user" height="40" />
                      <span className="text-ellipsis">Log in with NEAR</span>
                  </a>
              }
            </div>
          </div>
          <div className="near-dapp-body">
            <div>
              { currentUser && <Drops {...{currentUser, updateUser}} />}
            </div>
          </div>
        </div>
      </div>
    )
  }

}

export default App;
