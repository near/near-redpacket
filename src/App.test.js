import "regenerator-runtime/runtime";
import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

let near;
let contract;
let accountId;
let walletConnection
beforeAll(async function () {
    // NOTE: nearlib and nearConfig are made available by near-shell/test_environment
    console.log('nearConfig', nearConfig);
    near = await nearlib.connect(nearConfig);
    accountId = nearConfig.contractName;
    contract = await near.loadContract(nearConfig.contractName, {
        viewMethods: ['welcome'],
        changeMethods: [],
        sender: accountId
    });

    // Fake instance of WalletConnection
    // Feel free to modify for specific tests
    walletConnection = {
      requestSignIn() {
      },
      signOut() {
      },
      isSignedIn() {
        return true;
      },
      getAccountId() {
        return accountId;
      }
    }
});

it('renders without crashing', () => {
  const app = renderer.create(<App contract={contract} wallet={walletConnection}/>);
  let tree = app.toJSON();
  expect(tree).toMatchSnapshot();
});
