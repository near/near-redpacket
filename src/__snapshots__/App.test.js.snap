// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="App-header"
>
  <div
    className="image-wrapper"
  >
    <img
      alt="NEAR logo"
      className="logo"
      src=""
    />
    <p>
      <span
        aria-label="fish"
        role="img"
      >
        🐟
      </span>
       NEAR protocol is a new blockchain focused on developer productivity and useability!
      <span
        aria-label="fish"
        role="img"
      >
        🐟
      </span>
    </p>
    <p>
      <span
        aria-label="chain"
        role="img"
      >
        ⛓
      </span>
       This little react app is connected to blockchain right now. 
      <span
        aria-label="chain"
        role="img"
      >
        ⛓
      </span>
    </p>
    <p
      style={
        Object {
          "color": "#0072CE",
          "fontSize": "1.5rem",
          "textShadow": "1px 1px #D1CCBD",
        }
      }
    />
  </div>
  <div>
    <div>
      <button
        onClick={[Function]}
      >
        Log out
      </button>
      <button
        onClick={[Function]}
      >
        Change greeting
      </button>
    </div>
  </div>
  <div>
    <div
      className="logo-wrapper"
    >
      <img
        alt="logo"
        className="App-logo margin-logo"
        src=""
      />
      <img
        alt="logo"
        className="App-logo"
        src=""
      />
    </div>
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
    <p>
      <span
        aria-label="net"
        role="img"
      >
        🕸
      </span>
       
      <a
        className="App-link"
        href="https://nearprotocol.com"
      >
        NEAR Website
      </a>
       
      <span
        aria-label="net"
        role="img"
      >
        🕸
      </span>
    </p>
    <p>
      <span
        aria-label="book"
        role="img"
      >
        📚
      </span>
      <a
        className="App-link"
        href="https://docs.nearprotocol.com"
      >
         Learn from NEAR Documentation
      </a>
       
      <span
        aria-label="book"
        role="img"
      >
        📚
      </span>
    </p>
  </div>
</div>
`;
